lib/Mongo/MongoDB.php

```diff
@@ -130,7 +132,13 @@ public function getCollectionInfo(array $options = [])
             unset($options['includeSystemCollections']);
         }
 
-        $collections = $this->db->listCollections($options);
+        try {
+            $collections = $this->db->listCollections($options);
+        } catch (\MongoDB\Driver\Exception\ExecutionTimeoutException $e) {
```

Comment:
I know the ExceptionConverter converts this to a `MongoCursorException` - I'd actually prefer converting it to `MongoExecutionTimeoutException` and adding special handling for this in the cursor class.